crates/websocket/src/send.rs

```diff
@@ -43,3 +51,72 @@ impl SendWebsocketMessage for PostId {
     Ok(res)
   }
 }
+
+#[async_trait::async_trait(?Send)]
+pub trait SendCommentWebsocketMessage {
+  type ResponseType;
+  async fn send_ws_message_simple<OP: ToString + Send + OperationType + 'static>(
+    self,
+    op: OP,
+    context: &LemmyContext,
+  ) -> Result<Self::ResponseType, LemmyError>;
+  async fn send_ws_message<OP: ToString + Send + OperationType + 'static>(
+    self,
+    op: OP,
+    websocket_id: Option<ConnectionId>,
+    form_id: Option<String>,
+    person_id: Option<PersonId>,
+    recipient_ids: Vec<LocalUserId>,
+    context: &LemmyContext,
+  ) -> Result<Self::ResponseType, LemmyError>;
+}
+
+// TODO: in many call sites in apub crate, we are setting an empty vec for recipient_ids,
+//       we should get the actual recipient actors from somewhere
+#[async_trait::async_trait(?Send)]
+impl SendCommentWebsocketMessage for CommentId {
```

Comment:
These to me are strange, like mixing functional and OO paradigms. 

The main benefit of trait is that you're defining verbs for similar types (both dogs and cats can jump), and that the type (dog or cat) has a lot of the information you need that you get to access to calculate the jump. The trait functions should either use `self`, or very few environmental variables. 

So something like `CommentRes.send(context)` is the only place I can think of here that this would make sense, otherwise its best to stick with the more clear and portable function paradigm.

Reply:
Side point but I rewatched [this](https://www.youtube.com/watch?v=0if71HOyVjY) on functional programming recently:  and it definitely made me think about being more precise when deciding when to use the Object vs functions paradigm.